e2e-test/e2e.py

```diff
@@ -96,6 +99,71 @@ def test_post_resource_increase_count(
         "%s %ss returned for %s", value_from_proxy, resource_search_pair[0], patient_id,
     )
 
+@patch.dict(os.environ, {"AUDIT_EVENT_LOGGING_ENABLED_ENV": "true"})
+def test_post_resource_with_logging_enabled_creates_audit_event(
+    file_name: str,
+    hapi: clients.HapiClient,
+    fhir_proxy: clients.FhirProxyClient,
+    auth: clients.AuthClient,
+) -> None:
+    """Test to add a resource to the backend via the Proxy and verify AuditEvent creation."""
+    token = auth.get_auth_token()
+ 
+    initial_audit_event_count = hapi.get_audit_event_count()
+    logging.info("Initial AuditEvent count: %d", initial_audit_event_count)
+
+    fhir_proxy.post_resource("Observation", file_name, token)
+    logging.info("Posted Observation resource via Proxy.")
+
+    max_wait_seconds = 60
+    poll_interval = 5
+    waited = 0
+    while waited < max_wait_seconds:
+        current_audit_event_count = hapi.get_audit_event_count()
+        if current_audit_event_count == initial_audit_event_count + 1:
```

Comment:
In addition to check the number of resources, please check some fields of the created AuditEvent. As a minimum, in the case of `POST`, I like to check the reference to the created resource with proper ID (to make sure that the returned ID is properly processed). On the other extreme, you can add the expected AutidEvent resource in a test `json` file and compare the two resources.